src/sentry/utils/email/message_builder.py

```diff
@@ -155,17 +153,15 @@ def build(
 
         subject = force_text(self.subject)
 
-        if self.reply_reference is not None:
-            reference = self.reply_reference
+        if isinstance(self.reference, Activity):
+            self.reference = self.reference.group
```

Comment:
@mgaeta It seems weird that we would cause this kind of side effect in a build method. This has the end result that if you call build twice for an activity event, you'd get two different results because the reference changed to a group. Can we call build without modifying the underlying message builder object?